triggers/service/src/test-suite/scala/com/daml/lf/engine/trigger/CliSpec.scala

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf.engine.trigger
+
+import akka.http.scaladsl.model.Uri
+import com.daml.auth.middleware.api.{Client => AuthClient}
+import com.daml.bazeltools.BazelRunfiles.requiredResource
+import com.daml.lf.speedy.Compiler
+import com.daml.platform.services.time.TimeProviderType
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AsyncWordSpec
+import pureconfig.error.{CannotReadFile, ConfigReaderFailures}
+
+import java.nio.file.Paths
+import java.util.concurrent.TimeUnit
+import scala.concurrent.duration.FiniteDuration
+
+class CliSpec extends AsyncWordSpec with Matchers {
+  val confFile = "triggers/service/src/test-suite/resources/trigger-service.conf"
+  def loadCli(file: String): Cli = {
+    Cli.parse(Array("--config", file), Set()).getOrElse(fail("Could not load Cli on parse"))
+  }
+
+  "should pickup the config file provided" in {
+    val file = requiredResource(confFile)
+    val cli = loadCli(file.getAbsolutePath)
+    cli.configFile should not be empty
+  }
+
+  "should be able to successfully load the config based on the file provided" in {
+    val file = requiredResource(confFile)
+    val cli = loadCli(file.getAbsolutePath)
+    cli.configFile should not be empty
+    cli.loadFromConfigFile match {
+      case Left(ex) => fail(s"Failed to load config from config file: ${ex.head.description}")
+      case Right(c) =>
+        c.address shouldBe "127.0.0.1"
+        c.port shouldBe Cli.DefaultHttpPort
+        c.portFile shouldBe Some(Paths.get("port-file"))
```

Comment:
I get that you're trying to improve the reporting, but that comes at the cost of getting no compiler indication that this test should be updated should a new field be added to `Cli`. It would be better to have a single mega equality test for that reason, despite the reporting being worse.

There's an easy way to explicitly exclude a field that you don't want to compare; for example, if you don't want to compare `c.initDb`, you could just write

```scala
c.copy(initDb = false) should ===(Cli(all, other, props, initDb = false))
```

with the presence in the `copy` call effectively being the signal that you want its comparison to vacuously pass.